Userland/Libraries/LibAudio/ConnectionFromClient.cpp

```diff
@@ -62,6 +62,9 @@ ErrorOr<void> ConnectionFromClient::async_enqueue(FixedArray<Sample>&& samples)
     update_good_sleep_time();
     m_user_queue->append(move(samples));
     // Wake the background thread to make sure it starts enqueuing audio.
+    // FIXME: This can lead to a deadlock if the thread has gone to sleep but didn't set the flag yet.
+    //        If our caller depends on samples being played before it calls async_enqueue again (as is usually the case),
+    //        it will never call into this function again and we don't get a chance to actually wake the enqueuer thread.
```

Comment:
Your explanation in the commit message raises the question of why we don't set an event queue wake regardless of whether the thread is awake or not. Is this intended as an optimization?

Reply:
Yes. If we send a wake every time, those just queue up as the event loop doesn't run as long as the enqueuer thread isn't finished playing audio. Therefore, as soon as we get a single break of audio, there would be a huge spike in useless calls into the custom event handler from all the queued event.